src/main/java/no/nav/pto/veilarbportefolje/aktiviteter/AktiviteterRepositoryV2.java

```diff
@@ -106,11 +116,15 @@ public void setTilFullfort(String aktivitetid) {
         db.update("UPDATE aktiviteter SET status = 'fullfort' WHERE aktivitetid = ?", aktivitetid);
     }
 
-    public List<Moteplan> hentFremtidigeMoter(VeilederId veilederIdent, EnhetId enhet) {
+    public List<Moteplan> hentFremtidigeMoter(VeilederId veilederIdent, EnhetId enhet, boolean tilgangTilKode6, boolean tilgangTilKode7, boolean tilgangTilEgenAnsatt) {
         List<Moteplan> result = new ArrayList<>();
+        String skjermetDiskresjonskoder = hentSkjermeteDiskresjonskoder(tilgangTilKode6, tilgangTilKode7);
```

Comment:
this logic about skjerming is maybe too much for one function, I would suggest somehow to devide that into smaller parts

Reply:
done 👍